tests/components/buienradar/test_camera.py

```diff
@@ -0,0 +1,172 @@
+"""The tests for generic camera component."""
+import asyncio
+
+from homeassistant.util import dt as dt_util
+
+from homeassistant.setup import async_setup_component
+
+# An infinitesimally small time-delta.
+EPSILON_DELTA = 0.0000000001
+
+
+def radar_map_url(dim: int = 512) -> str:
+    """Build map url, defaulting to 512 wide (as in component)."""
+    return ("https://api.buienradar.nl/"
+            "image/1.0/RadarMapNL?w={dim}&h={dim}").format(dim=dim)
+
+
+@asyncio.coroutine
+def test_fetching_url_and_caching(aioclient_mock, hass, hass_client):
+    """Test that it fetches the given url."""
+    aioclient_mock.get(radar_map_url(), text='hello world')
+
+    yield from async_setup_component(hass, 'camera', {
```

Comment:
See above.

Reply:
Cargo cult copied this. Prefer this style and will change it.